pkg/updatestrategy/aws_asg.go

```diff
@@ -317,25 +320,51 @@ func (n *ASGNodePoolsBackend) Terminate(node *Node, decrementDesired bool) error
 
 	_, err := n.asgClient.TerminateInstanceInAutoScalingGroup(params)
 	if err != nil {
-		// could be because the instance is already being terminated, check for that
-		status, serr := n.ec2Client.DescribeInstanceStatus(&ec2.DescribeInstanceStatusInput{
-			IncludeAllInstances: aws.Bool(true),
-			InstanceIds:         []*string{aws.String(instanceId)},
-		})
+		_, serr := n.instanceState(instanceId)
+		if serr != nil {
+			return fmt.Errorf("failed to terminate instance '%s': %v, %v", instanceId, err, serr)
+		}
+	}
 
-		if serr != nil || len(status.InstanceStatuses) == 0 {
-			return err
+	// wait for the instance to be terminated/stopped
+	instanceState := func() error {
+		state, err := n.instanceState(instanceId)
+		if err != nil {
+			return backoff.Permanent(err)
 		}
 
-		switch aws.StringValue(status.InstanceStatuses[0].InstanceState.Name) {
-		case ec2.InstanceStateNameShuttingDown, ec2.InstanceStateNameStopping, ec2.InstanceStateNameTerminated, ec2.InstanceStateNameStopped:
+		switch state {
+		case ec2.InstanceStateNameShuttingDown, ec2.InstanceStateNameStopping:
+			return errors.New("instance shutting down")
+		case ec2.InstanceStateNameTerminated, ec2.InstanceStateNameStopped:
 			return nil
 		default:
-			return err
+			return fmt.Errorf("unexpected instance state '%s'", state)
```

Comment:
Should this be `backoff.Permanent`?

Reply:
If there's a default backoff limit, then we could just ignore. It's not like this is going to happen often (or ever), I guess.